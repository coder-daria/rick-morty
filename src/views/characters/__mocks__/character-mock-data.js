export const LOCATION_NAME = 'Earth';

export const CHARACTER_NOT_FORMATED_MOCKED_DETAILS = {
  location: {
    name: LOCATION_NAME,
  },
  origin: {
    name: LOCATION_NAME,
  },
};

export const CHARACTER_FORMATED_MOCKED_DETAILS = {
  location: LOCATION_NAME,
  origin: LOCATION_NAME,
};

export const CHARACTER_DETAILED_MOCKED_DATA = {
  created: '11/25/2020, 4:59:08 PM',
  episode: [],
  gender: 'Male',
  image: '',
  name: 'Rick Sanchez',
  species: 'Alien',
  status: 'Alive',
  type: '',
};
