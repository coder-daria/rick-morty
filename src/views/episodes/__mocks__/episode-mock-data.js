export const EPISODE_NOT_FORMATED_MOCKED_DETAILS = {
  air_date: '11/25/2020, 4:59:08 PM',
};

export const EPISODE_FORMATED_MOCKED_DETAILS = {
  'air date': '11/25/2020, 4:59:08 PM',
};

export const EPISODE_DETAILS_MOCKED_DATA = {
  name: 'Sunny day',
  episode: '0E12',
  created: '11/25/2019, 4:59:08 PM',
  characters: [{ name: 'Tom' }],
};
