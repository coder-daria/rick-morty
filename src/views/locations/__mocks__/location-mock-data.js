export const LOCATION_DETAILS_MOCKED_DATA = {
  name: 'Earth',
  created: '11/25/2019',
  dimension: '',
  type: '',
  residents: [{ name: 'Tom' }],
};
