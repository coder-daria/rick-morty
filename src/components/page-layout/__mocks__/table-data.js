import { CHARACTER_NOT_FORMATED_MOCKED_DETAILS } from '../../../views/characters/__mocks__/character-mock-data';

export const PAGE_INFO = {
  pages: 1,
  count: 1,
  next: null,
  prev: null,
};

export const TABLE_DATA = {
  columns: [
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
    },
    {
      title: 'Species',
      dataIndex: 'species',
      key: 'species',
    },
    {
      title: 'Origin',
      dataIndex: 'origin',
      key: 'origin',
    },
    {
      title: 'Location',
      dataIndex: 'location',
      key: 'location',
    },
  ],
  loading: false,
  pageInfo: PAGE_INFO,
  tableData: [
    {
      id: 1,
      name: 'Rick Sanchez',
      species: 'Alien',
      ...CHARACTER_NOT_FORMATED_MOCKED_DETAILS,
    },
  ],
  toggleDrawer: jest.fn(),
  setCurrentPage: jest.fn(),
};
