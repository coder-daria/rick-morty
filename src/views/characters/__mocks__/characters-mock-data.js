import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';
import { CHARACTER_NOT_FORMATED_MOCKED_DETAILS } from './character-mock-data';

const CHARACTER_TABLE_MOCKED_DATA = {
  id: 1,
  name: 'Rick Sanchez',
  species: 'Alien',
  ...CHARACTER_NOT_FORMATED_MOCKED_DETAILS,
};

export const CHARACTERS_MOCKED_DATA = {
  characters: {
    results: [CHARACTER_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
