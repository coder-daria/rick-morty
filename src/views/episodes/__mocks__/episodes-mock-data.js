import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';

const EPISODE_TABLE_MOCKED_DATA = {
  characters: [{ name: 'Tom' }],
  episode: '0E12',
  id: 1,
  name: 'Sunny day',
};

export const EPISODES_MOCKED_DATA = {
  episodes: {
    results: [EPISODE_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
