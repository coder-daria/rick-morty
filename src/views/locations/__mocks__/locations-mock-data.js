import { PAGE_INFO } from '../../../components/page-layout/__mocks__/table-data';

const LOCATION_TABLE_MOCKED_DATA = {
  id: 1,
  created: '11/25/2019, 4:59:08 PM',
  dimension: '',
  name: 'Sunny day',
  type: '',
  residents: [{ name: 'Tom' }],
};

export const LOCATIONS_MOCKED_DATA = {
  locations: {
    results: [LOCATION_TABLE_MOCKED_DATA],
    info: PAGE_INFO,
  },
};
