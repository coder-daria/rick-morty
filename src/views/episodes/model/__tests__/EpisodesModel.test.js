import { EPISODES_MOCKED_DATA } from '../../__mocks__/episodes-mock-data';
import getEpisodesData from '../EpisodesModel';

const {
  episodes: { results, info },
} = EPISODES_MOCKED_DATA;

const [FIRST_CHARACTER] = results;

describe('CharactersModel', () => {
  it('should return episodes data', () => {
    const result = getEpisodesData(EPISODES_MOCKED_DATA, jest.fn());

    expect(result).toMatchSnapshot({
      episodes: [
        {
          ...FIRST_CHARACTER,
          characters: FIRST_CHARACTER.characters.length,
        },
      ],
      pageInfo: info,
    });
  });
});
