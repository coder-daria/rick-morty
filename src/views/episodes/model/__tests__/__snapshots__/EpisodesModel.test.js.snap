// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersModel should return episodes data 1`] = `
Object {
  "episodes": Array [
    Object {
      "characters": 1,
      "episode": "0E12",
      "id": 1,
      "name": "Sunny day",
    },
  ],
  "pageInfo": Object {
    "count": 1,
    "next": null,
    "pages": 1,
    "prev": null,
  },
}
`;
